packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -104,7 +138,14 @@ class DependencyExtractionWebpackPlugin {
 		return JSON.stringify( asset );
 	}
 
+	/** @type {webpack.WebpackPluginInstance['apply']} */
 	apply( compiler ) {
+		this.useModules = Boolean( compiler.options.output?.module );
+
+		if ( this.useModules ) {
+			this.externalsPlugin.type = 'module';
+		}
```

Comment:
This isn't necessary at all, but it may be helpful for folks to know whether they've compiled for modules or not.

Otherwise, the output could be identical. Webpack can be very difficult to debug, so seeing this in the output may be helpful.